integration-tests/file/src/test/java/org/apache/camel/quarkus/component/file/it/FileTest.java

```diff
@@ -17,48 +17,150 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 import io.restassured.path.json.JsonPath;
 import io.restassured.response.ValidatableResponse;
+import org.hamcrest.Matcher;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 
+import static org.apache.camel.quarkus.component.file.it.FileResource.CONSUME_BATCH;
+import static org.apache.camel.quarkus.component.file.it.FileResource.SEPARATOR;
+import static org.apache.camel.quarkus.component.file.it.FileResource.SORT_BY;
 import static org.awaitility.Awaitility.await;
 import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNot.not;
 
 @QuarkusTest
 class FileTest {
 
     private static final String FILE_BODY = "Hello Camel Quarkus";
+    private static final String FILE_CONTENT_01 = "Hello1";
+    private static final String FILE_CONTENT_02 = "Hello2";
+    private static final String FILE_CONTENT_03 = "Hello3";
+    private static final String FILE_BODY_UTF8 = "Hello World \u4f60\u597d";
+
+    private List<Path> pathsToDelete = new LinkedList<>();
+
+    @AfterEach
+    public void afterEach() {
+        pathsToDelete.stream().forEach(p -> {
+            try {
+                Files.delete(p);
+            } catch (IOException e) {
+                //ignore
+            }
+        });
+
+        pathsToDelete.clear();
+    }
 
     @Test
     public void file() {
         // Create a new file
-        String fileName = RestAssured.given()
-                .contentType(ContentType.TEXT)
-                .body(FILE_BODY)
-                .post("/file/create/in")
-                .then()
-                .statusCode(201)
-                .extract()
-                .body()
-                .asString();
+        String fileName = createFile(FILE_BODY, "/file/create/in");
 
         // Read the file
         RestAssured
-                .get("/file/get/in/" + Paths.get(fileName).getFileName())
+                .post("/file/get/in/" + Paths.get(fileName).getFileName())
                 .then()
                 .statusCode(200)
                 .body(equalTo(FILE_BODY));
     }
 
+    @Test
+    public void consumerCharset() throws UnsupportedEncodingException {
+        // Create a new file
+        createFile(FILE_BODY_UTF8, "/file/create/charsetUTF8", "UTF-8", null);
+        createFile(FILE_BODY_UTF8, "/file/create/charsetISO", "UTF-8", null);
+
+        await().atMost(10, TimeUnit.SECONDS).until(() -> getFromMock("charsetUTF8", equalTo(FILE_BODY_UTF8)));
+
+        await().atMost(10, TimeUnit.SECONDS).until(() -> {
+            // File content read as ISO-8859-1 has to have different content (because file contains some unknown characters)
+            return getFromMock("charsetISO", not(equalTo(FILE_BODY_UTF8)));
```

Comment:
I think we can be a bit more specific about what we expect here. We want to see that the given encoding was applied correctly. If we used characters that make sense in both UTF8 and ISO latin 2, such as ľščťžýďáíéäňô in FILE_BODY_UTF8, we could assert that the binary content is "ľščťžýďáíéäňô".getBytes("iso-8859-2")

Reply:
I'll refactor it

Reply:
done